app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,69 @@
+class CompaniesController < ApplicationController
+  before_action :set_company, only: %i[ show edit update destroy ]
+
+  # GET /companies or /companies.json
+  def index
+    @companies = Company.all
+  end
+
+  # GET /companies/1 or /companies/1.json
+  def show
+  end
+
+  # GET /companies/new
+  def new
+    @company = Company.new
+  end
+
+  # GET /companies/1/edit
+  def edit
+  end
+
+  # POST /companies or /companies.json
+  def create
+    @company = Company.new(company_params)
+
+    respond_to do |format|
+      if @company.save
+        format.html { redirect_to @company, notice: "Company was successfully created." }
+        format.json { render :show, status: :created, location: @company }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /companies/1 or /companies/1.json
+  def update
+    respond_to do |format|
+      if @company.update(company_params)
+        format.html { redirect_to @company, notice: "Company was successfully updated." }
+        format.json { render :show, status: :ok, location: @company }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /companies/1 or /companies/1.json
+  def destroy
+    @company.destroy
+    respond_to do |format|
+      format.html { redirect_to companies_url, notice: "Company was successfully destroyed." }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
I recommend that they perform a logical deletion, that is, that the information in the database does not disappear, but is simply hidden from the end user.